pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/LawOfDemeterRule.java

```diff
@@ -84,7 +81,7 @@ private void addViolations(List<MethodCall> calls, RuleContext ctx) {
      * Collects the information of one identified method call. The method call
      * might be a violation of the Law of Demeter or not.
      */
-    private static class MethodCall {
+    public static class MethodCall {
```

Comment:
Is this necessary to change this class to public?
We should keep as much stuff private as possible. The reason is: As soon as we publish something, it becomes public API and it will be difficult to change without potentially breaking compatibility.

Note: This applies to all the other changes, where you increased the visibility of classes/methods.